### PR TITLE
Feature/LINK-1774 keyclock authentication

### PR DESCRIPTION
**Pull request by arpargo**
Replace authentication sollution with HDS Login component and Helsinki Tunnistus (Keycloak)

[LINK-1700](https://helsinkisolutionoffice.atlassian.net/browse/LINK-1700) & [LINK-1774](https://helsinkisolutionoffice.atlassian.net/browse/LINK-1774) 

[LINK-1700]: https://helsinkisolutionoffice.atlassian.net/browse/LINK-1700?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LINK-1774]: https://helsinkisolutionoffice.atlassian.net/browse/LINK-1774?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/domain/app/header/Header.tsx [hunk 1]
```diff
@@ -56,10 +56,24 @@ const Header: React.FC = () => {
...
+      if (authUser?.profile) {
+        const { name, family_name, given_name } = authUser?.profile;
+        if (name) setDisplayName(name);
+        else if (given_name && family_name) {
+          setDisplayName(`${given_name} ${family_name}`);
+        } else if (given_name) setDisplayName(given_name);
+        else setDisplayName('');
+      } else setDisplayName('');
+    }
+  }, [authUser, authenticated]);
```
Comment (jorilindell) [inline src/domain/app/header/Header.tsx:76]: This change breaks unit tests in `src/domain/app/header/__tests__/Header.test.tsx` file.
You can fix failing tests by changing
```
import {
  mockedRegistrationUserResponse,
  mockedRegularUserResponse,
  mockedUserResponse,
  mockedUserWithoutOrganizationsResponse,
  userFirstName,
} from '../../../user/__mocks__/user';

...

const findUserMenuButton = () =>
  screen.findByRole('button', { name: userFirstName }, { timeout: 10000 });
``` 
to
```
import {
  mockedRegistrationUserResponse,
  mockedRegularUserResponse,
  mockedUserResponse,
  mockedUserWithoutOrganizationsResponse,
  userName,
} from '../../../user/__mocks__/user';

...

const findUserMenuButton = () =>
  screen.findByRole('button', { name: userName }, { timeout: 10000 });
```

**Review by jorilindell — CHANGES_REQUESTED** [review]
Generally looks good. Added some improvement ideas. At least failing tests should be fixed

File: src/domain/app/header/Header.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (timwessman) [reply src/domain/app/header/Header.tsx]: Thanks, good small improvement

File: src/domain/auth/constants.ts [hunk 2]
```diff
@@ -16,14 +12,15 @@ export const loginProviderProps: LoginProviderProps = {
...
+    response_type: 'code',
   },
   apiTokensClientSettings: {
     url: import.meta.env.REACT_APP_OIDC_API_TOKENS_URL,
     queryProps: {
       grantType: 'urn:ietf:params:oauth:grant-type:uma-ticket',
       permission: '#access',
     },
-    audiences: [import.meta.env.REACT_APP_OIDC_API_SCOPE],
+    audiences: ['linkedcomponents-ui-dev', 'profile-api-dev'],
```
[the remark this answers is not in the document]
Comment (timwessman) [reply src/domain/auth/constants.ts]: Good, let's skip the profile api token

**Comment by timwessman** [comment]
Thanks for good PR comments, I commited the fixes

**Review by jorilindell — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -35,66 +35,13 @@ The web application is running at http://localhost:3000
...
-
 ### Install local Linked Events API
 
-Clone the repository (https://github.com/City-of-Helsinki/linkedevents). Follow the instructions for running linkedevents with docker. Before running `docker compose up` use the `env.example` template as base for `/docker/django/.env` and also set the following settings there:
-
-- TOKEN_AUTH_AUTHSERVER_URL=http://tunnistamo-backend:8000/openid
-- SOCIAL_AUTH_TUNNISTAMO_OIDC_ENDPOINT=http://tunnistamo-backend:8000/openid
-- SOCIAL_AUTH_TUNNISTAMO_KEY=linkedevents-api-admin
-- SOCIAL_AUTH_TUNNISTAMO_SECRET<linkedevents-api-admin client secret from tunnistamo>
+Clone the repository (https://github.com/City-of-Helsinki/linkedevents). Follow the instructions for running linkedevents with docker. Before running `docker compose up` use the `env.example` template as base for `/docker/django/.env`
```
Comment (jorilindell) [inline README.md:40]: Authentication between UI and API doesn't work with default env.example values. You could change this to 
```
Clone the repository (https://github.com/City-of-Helsinki/linkedevents). Follow the instructions for running linkedevents with docker. Before running `docker compose up` use the `env.example` template as base for `/docker/django/.env` and add following variables:

- TOKEN_AUTH_ACCEPTED_AUDIENCE=linkedevents-api-dev
- TOKEN_AUTH_AUTHSERVER_URL=https://tunnistus.test.hel.ninja/auth/realms/helsinki-tunnistus
```

**Pull request merged**